tests/extract-algorithms.js

```diff
@@ -56,6 +56,24 @@ const tests = [
       }
     ]
   },
+  {
+    title: 'extracts one-paragraph algorithms',
+    html: `
+      <section>
+        <p>To <dfn data-export data-dfn-type=dfn id="algo-id">do nothing</dfn>, keep calm and carry on.</p>
+      </section>`,
+    algorithms: [
+      {
+        name: 'do nothing',
+        href: 'about:blank#algo-id',
+        html: 'To <dfn data-export="" data-dfn-type="dfn" id="algo-id">do nothing</dfn>, keep calm and carry on.',
+        rationale: '"To <dfn>"',
+        steps: [ {
+	  html: 'To <dfn data-export="" data-dfn-type="dfn" id="algo-id">do nothing</dfn>, keep calm and carry on.'
```

Comment:
Conceptually, do we really want to duplicate the text both in the root `html` property, and as a single step? Wouldn't it be clearer to not have any `steps` property in such cases? I.e., to end up with:

```js
    [
      {
        name: 'do nothing',
        href: 'about:blank#algo-id',
        html: 'To <dfn data-export="" data-dfn-type="dfn" id="algo-id">do nothing</dfn>, keep calm and carry on.',
        rationale: '"To <dfn>"'
     }
    ]
```

Reply:
I wondered about it, but started with the smallest code change. I think the options are:
* split the actual procedural stuff ("keep calm and carry on" into the step (ideal, but maybe hard to do robustly)
* not have steps (which at least provide a simple signal this is a "short form" algorithm)
* duplicate root markup in a single step

I'm happy to use the 2nd option at least as a starting point, which I agree would be an improvement.

Reply:
+1 for the second option. It  has the advantage of being consistent with the way a step in an algorithm is extracted when it does not define substeps: It gets reported with an `html` property and no `steps`.

Reply:
done in [8487c8e](https://github.com/w3c/reffy/pull/1614/commits/8487c8e5a2902bbdee40dd535daa101381516c43)